From da81b3b6f6de70409af8d09ca6549a3dfcd78336 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Mon, 30 Jul 2012 22:22:24 +0800
Subject: [PORT FROM MAIN]Fix rtsp streaming continuously seeking, video freeze issue.

BZ: 49305

After seeking, the video rtcp packet receives too late, and the rtp packets
which receives before the rtcp packets are cached, if we don't clear these packets,
when finish the nextseeking it will use the cache rtp packets to calculate the
timestamp and it will get a wrong timestamp for A/V sync.

orig-patch:68627:(BZ:35189)Fix rtsp streaming continuously seeking, video freeze issue.

Change-Id: Ia46fb3c286c6ffa8cbfdc12b21090727e318b997
Orig-Change-Id: Ic3273beb069765100569908aef9da6593ba0885a
Signed-off-by: ywan171 <yi.a.wang@intel.com>
---
 media/libstagefright/rtsp/MyHandler.h |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/rtsp/MyHandler.h b/media/libstagefright/rtsp/MyHandler.h
index e067e20..ed8ad53 100644
--- a/media/libstagefright/rtsp/MyHandler.h
+++ b/media/libstagefright/rtsp/MyHandler.h
@@ -1182,6 +1182,8 @@ struct MyHandler : public AHandler {
 
                     info->mRTPAnchor = 0;
                     info->mNTPAnchorUs = -1;
+                    // clear the cached packets
+                    info->mPackets.clear();
                 }
 
                 mAllTracksHaveTime = false;
-- 
1.7.4.1

